test/puppet/puppet_legacy_runtime_configuration_test.rb

```diff
@@ -0,0 +1,87 @@
+  def setup
+    @configuration = ::Proxy::PuppetLegacy::PluginConfiguration.new
+  end
+
+  def test_use_future_parser_when_main_parser_is_set
+    assert @configuration.use_future_parser?(:main => {:parser => "future"})
+  end
+
+  def test_use_future_parser_when_master_parser_is_set
+    assert @configuration.use_future_parser?(:master => {:parser => "future"})
+  end
+
+  def test_use_future_parser_when_main_and_master_are_missing
+    assert !@configuration.use_future_parser?({})
+  end
+
+  def test_use_environment_api_when_main_environmentpath_is_set
+    assert @configuration.use_environment_api?(:main => {:environmentpath => ["a/path"]})
+  end
+
+  def test_use_environment_api_when_master_environmentpath_is_set
+    assert @configuration.use_environment_api?(:master => {:environmentpath => ["a/path"]})
+  end
+
+  def test_use_environment_api_when_main_and_master_are_missing
+    assert !@configuration.use_environment_api?({})
+  end
+
+  def test_puppet_conf_exists_is_used_in_load_programmable_settings_call
+    @configuration.expects(:puppet_conf_exists?).with("a/path")
+    @configuration.stubs(:load_puppet_configuration).returns({})
+    @configuration.load_programmable_settings(:puppet_conf => "a/path")
+  end
+
+  class LegacyProviderConfigurationForTesting < ::Proxy::PuppetLegacy::PluginConfiguration
+    def initialize(puppet_config)
+      @config = puppet_config
+    end
+
+    def puppet_conf_exists?(a_path); end
+  end
+
+  def test_load_puppet_configuration_is_used_in_load_programmable_settings_call
+    configuration = LegacyProviderConfigurationForTesting.new({})
+    configuration.stubs(:load_puppet_configuration).with("a/path").returns({})
+    configuration.load_programmable_settings(:puppet_conf => "a/path")
+  end
+
+  def test_load_programmable_settings_sets_cached_future_parser
+    configuration = LegacyProviderConfigurationForTesting.new(:main => {:parser => "future"})
+    assert_equal :cached_future_parser, configuration.load_programmable_settings(:use_cache => true)[:classes_retriever]
+  end
+
+  def test_load_programmable_settings_sets_future_parser
+    configuration = LegacyProviderConfigurationForTesting.new(:main => {:parser => "future"})
+    assert_equal :future_parser, configuration.load_programmable_settings(:use_cache => false)[:classes_retriever]
+  end
+
+  def test_load_programmable_settings_sets_cached_legacy_parser
+    configuration = LegacyProviderConfigurationForTesting.new({})
+    assert_equal :cached_legacy_parser, configuration.load_programmable_settings(:use_cache => true)[:classes_retriever]
+  end
+
+  def test_load_programmable_settings_sets_legacy_parser
+    configuration = LegacyProviderConfigurationForTesting.new({})
+    assert_equal :legacy_parser, configuration.load_programmable_settings(:use_cache => false)[:classes_retriever]
+  end
+
+  def test_load_programmable_settings_sets_config_file_for_environments_retriever_for_puppet_before_3_2
```

Comment:
I think it's missing a test case for using `:api_v2` with `:puppet_version => "3.2"` and no override.